### PR TITLE
fix: protect_reference column > VARCHAR(255)

### PR DESCRIPTION
**Pull request by SPlanzer**
modifies column "protect_reference" from VARCHAR(100) to VARCHAR(255) as per https://github.com/linz/linz-bde-uploader/issues/279

### CODE REVIEW

File: sql/99-patches.sql [hunk 1]
```diff
@@ -294,5 +294,74 @@ $P$
...
+             FROM pg_catalog.pg_proc p, pg_catalog.pg_namespace n
+             WHERE p.proname = 'ver_is_table_versioned'
+             AND n.oid = p.pronamespace
+             AND n.nspname = 'table_version')
+  THEN
+    v_isversioned := true;
+  END IF;
+
+  -- Alter bde.crs_ttl_inst_protect
+  ALTER TABLE bde.crs_ttl_inst_protect
```
Comment (palmerj) [inline sql/99-patches.sql:324]: ver_versioned_table_change_column_type will change the main table column data type for you. So this should only be run if that table is not versioned.

File: sql/99-patches.sql [hunk 2]
```diff
@@ -294,5 +294,74 @@ $P$
...
+          'bde',
+          'crs_ttl_inst_protect',
+          'protect_reference',
+          'VARCHAR(255)'
+      );
+    END IF;
+  END IF;
+
+  -- Alter bde.crs_title and associated versioning table
+  ALTER TABLE bde.crs_title
```
Comment (palmerj) [inline sql/99-patches.sql:343]: As above

File: sql/99-patches.sql [hunk 3]
```diff
@@ -294,5 +294,74 @@ $P$
...
 
+--------------------------------------------------------------------------------
+-- protect_reference column VARCHAR(100) > VARCHAR(255)
+--
+-- For tables bde.crs_title & bde.crs_ttl_inst_protect
+-- as per https://github.com/linz/linz-bde-uploader/issues/279
+--------------------------------------------------------------------------------
+
+PERFORM _patches.apply_patch(
+    'protect_reference column VARCHAR(100) > VARCHAR(255)',
```
Comment (palmerj) [inline sql/99-patches.sql:305]: We should add a version prefix to this patch comment. I see @strk is using LOL version for the last few. Before that we used the up-coming BDE processor release number.

Comment (billgeo) [reply sql/99-patches.sql:305]: Last BDE deployment version was 1.9

Comment (billgeo) [reply sql/99-patches.sql:305]: So I guess next one should be 1.9.1 for a bug fix?

Comment (SPlanzer) [reply sql/99-patches.sql:305]: next version of BDE processor is 1.10

Comment (SPlanzer) [reply sql/99-patches.sql:305]: Or should we use the bde-schema version (1.12.1 or more likely 1.13)

**Comment by strk** [comment]
Why do we limit the VARCHAR fields at all ? The type is of variable lenght anyway, and we always only mean to import that data from BDE exports...

File: sql/99-patches.sql (hunk 3 above)
Comment (strk) [reply sql/99-patches.sql:305]: Jeremy refers to LOL version, not BDE version. I guess we could use the version of LOL which first happened to contain a value longer than 100 for that title reference

**Comment by strk** [comment]
I've pushed a commit adding automated testcase for being able to store long text (255 chars) into the relevant fields (base and revision tables). Test should fail when reverting the schema patch changes.

**Comment by palmerj** [comment]
> Why do we limit the VARCHAR fields at all ? The type is of variable lenght anyway, and we always only mean to import that data from BDE exports...

Informix has it defined this way, so it should be no problem. But we do this because we want to keep the constraint and metadata for the end users of the data in LDS. For some database engines the length is still important for storage. In some cases the length also helps with display of the columns (e.g Esri ArcMap table column widths)

File: sql/99-patches.sql (hunk 3 above)
Comment (billgeo) [reply sql/99-patches.sql:305]: So `LOL 3.22b: ` then? Seems okay to me.

Note: the last patch looks like it had a typo `LOL 3.22p: `

**Comment by strk** [comment]
I've renamed the patch to:

> LOL 3.22: Enlarge crs_title and crs_ttl_inst_protect protect_reference column to 255 chars

The '3.22b' is not a typo but the version as found in the BDE files
received when having the LTA added.

**Comment by strk** [comment]
Merged

**Comment by strk** [comment]
Merged with 12df32d1316531260c24fa0dd23e973a6f66973e

**Pull request closed without merging**